src/client_side_request.cc

```diff
@@ -106,11 +106,6 @@ static void checkFailureRatio(err_type, hier_code);
 
 ClientRequestContext::~ClientRequestContext()
 {
-    /*
-     * Release our "lock" on our parent, ClientHttpRequest, if we
-     * still have one
-     */
```

Comment:
This comment was unnecessary (because the corresponding code is self-documenting) and would have become misleading (because we now always have that "lock" -- this PR removes all `cbdataReferenceDone()` calls that could "unlock").